src/core/FaceLandmarker.tsx

```diff
@@ -0,0 +1,49 @@
+import * as React from 'react'
+import { createContext, ReactNode, useContext, useEffect, useState } from 'react'
+import { FilesetResolver, FaceLandmarker as FaceLandmarkerImpl, FaceLandmarkerOptions } from '@mediapipe/tasks-vision'
+
+const FaceLandmarkerContext = createContext({} as FaceLandmarkerImpl | undefined)
+
+type FaceLandmarkerProps = {
+  basePath?: string
+  options?: FaceLandmarkerOptions
+  children?: ReactNode
+}
+
+export const FaceLandmarkerDefaults = {
+  basePath: 'https://cdn.jsdelivr.net/npm/@mediapipe/tasks-vision@0.10.0/wasm',
+  options: {
+    baseOptions: {
+      modelAssetPath:
+        'https://storage.googleapis.com/mediapipe-models/face_landmarker/face_landmarker/float16/1/face_landmarker.task',
+      delegate: 'GPU',
+    },
+    runningMode: 'VIDEO',
+    outputFaceBlendshapes: true,
+    outputFacialTransformationMatrixes: true,
+  } as FaceLandmarkerOptions,
+}
+
+export function FaceLandmarker({
+  basePath = FaceLandmarkerDefaults.basePath,
+  options = FaceLandmarkerDefaults.options,
+  children,
+}: FaceLandmarkerProps) {
+  const [faceLandmarker, setFaceLandmarker] = useState<FaceLandmarkerImpl>()
+  useEffect(() => {
+    let ret: FaceLandmarkerImpl
+
+    FilesetResolver.forVisionTasks(basePath)
+      .then((vision) => FaceLandmarkerImpl.createFromOptions(vision, options))
+      .then((faceLandmarker) => setFaceLandmarker(faceLandmarker))
+      .catch((err) => console.error('error while creating faceLandmarker', err))
```

Comment:
I'd consider using suspense here.

Reply:
but how to deal with `ret.close()` cleanup with suspend?

Reply:
Ok, following [@drcmda additional advices on that](https://discord.com/channels/740090768164651008/741751532592038022/1114646670584578178), I've now converted it to suspend, but I let you check my impl is correct